portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,46 +31,95 @@
 
 
 
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   private ArrayList<String> messages = new ArrayList<String>();
 
+  static class Data{
+     Long id;
+     String comment;
+     long timestamp;
+
+     /**Constructor*/
+      public Data(long id, String comment, long timestamp) { 
+        this.id = id; 
+        this.comment= comment; 
+        this.timestamp= timestamp;
+    } 
+  }
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
       String commentString = request.getParameter("data-input");
+      //System.out.println(commentString);
+      long timestamp = System.currentTimeMillis();
+      //System.out.println(timestamp);
       messages.add(commentString);
+      //System.out.println(messages);
+      String maxstring = request.getParameter("comment-limit");
+      
+      //Create entity 
+      Entity dataEntity = new Entity("Data");
+      dataEntity.setProperty("comment", commentString);
+      dataEntity.setProperty("timestamp", timestamp);
+   
+      //Create datastore and put entity
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      if (commentString!=null && !commentString.isEmpty()) {datastore.put(dataEntity);}
+      
+
+      response.sendRedirect("index.html?comment-limit=" + maxstring);
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //response.setContentType("text/html;");
-    //response.getWriter().println("<h1>Hello Nnenna!</h1>");
-
-    //TO DO: Remove commented code.
-    // Convert the messages to JSON
-    //ArrayList<String> messages = new ArrayList<String>();
-    //messages.add("How are you?");
-    //messages.add("What did you have for breakfast?");
-    //messages.add("How old are you?");
-    String json = convertToJson(messages);
-
-    // Send the JSON as the response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
-  private String convertToJson(ArrayList<String> messages) {
+    //String json = convertToJson(messages);
+    Query query = new Query("Data").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
```

Comment:
You can limit the number of comments you retrieve from the datastore instead of iterating on the list manually.